core/collector.class.inc.php

```diff
@@ -716,32 +716,54 @@ public function Synchronize($iMaxChunkSize = 0)
 		$sResult = self::CallItopViaHttp("/synchro/synchro_exec.php?login_mode=$sLoginform",
 			$aData);
 
-		$iErrorsCount = 0;
+		$iErrorsCount = $this->ParseSynchroExecOutput($sResult);
+
+		return ($iErrorsCount == 0);
+	}
+
+	/**
+	 * Method that detects synchro exec errors and finds out details message
+	 * @param string $sResult: synchro_exec.php output
+	 *
+	 * @return int: error count
+	 * @throws \Exception
+	 */
+	public function ParseSynchroExecOutput($sResult) : int
+	{
 		if (preg_match_all('|<input type="hidden" name="loginop" value="login"|', $sResult, $aMatches)) {
 			// Hmm, it seems that the HTML output contains the login form !!
 			Utils::Log(LOG_ERR, "Failed to login to iTop. Invalid (or insufficent) credentials.");
 			$this->sErrorMessage .= "Failed to login to iTop. Invalid (or insufficent) credentials.\n";
-			$iErrorsCount = 1;
-		} else {
-			if (preg_match_all('/Objects (.*) errors: ([0-9]+)/', $sResult, $aMatches)) {
-				foreach ($aMatches[2] as $idx => $sErrCount) {
-					$iErrorsCount += (int)$sErrCount;
-					if ((int)$sErrCount > 0) {
-						Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' answered: {$aMatches[0][$idx]}");
-						$this->sErrorMessage .= $aMatches[0][$idx]."\n";
-					}
+			return 1;
+		}
+
+		$iErrorsCount = 0;
+		if (preg_match_all('/Objects (.*) errors: ([0-9]+)/', $sResult, $aMatches)) {
+			foreach ($aMatches[2] as $sDetailedMessage => $sErrCount) {
+				$iErrorsCount += (int)$sErrCount;
+				if ((int)$sErrCount > 0) {
+					Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' answered: {$aMatches[0][$sDetailedMessage]}");
+					$this->sErrorMessage .= $aMatches[0][$sDetailedMessage]."\n";
 				}
-			} else {
-				Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' failed.");
-				$this->sErrorMessage .= $sResult;
-				$iErrorsCount = 1;
 			}
+		} else {
+			Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' failed.");
+			$this->sErrorMessage .= $sResult;
+			return 1;
+		}
+
+		if (($iErrorsCount === 0) && preg_match('/<p>ERROR: (.*)\./', $sResult, $aMatches)) {
+			$sDetailedMessage = $aMatches[1];
+			Utils::Log(LOG_ERR, "Synchronization of data source '{$this->sSourceName}' answered: $sDetailedMessage");
+			$this->sErrorMessage .= $sDetailedMessage."\n";
+			return 1;
 		}
+
 		if ($iErrorsCount == 0) {
 			Utils::Log(LOG_INFO, "Synchronization of data source '{$this->sSourceName}' succeeded.");
 		}
 
-		return ($iErrorsCount == 0);
+		return $iErrorsCount;
 	}
 
 	public static function ParseSynchroOutput($sTrimmedOutput, $bDetailedOutput) : string
```

Comment:
What about this, if you want to make both methods be aligned in naming while keeping compatibility:
```suggestion
	/**
	 * @deprecated 1.3.1 N°6771
	 * @see static::ParseSynchroOutput
	 */
	public static function ParseSynchroOutput($sTrimmedOutput, $bDetailedOutput) : string {
		// log a deprecation message
		// Utils::Log(...);
		return static::ParseSynchroImportOutput($sTrimmedOutput, $bDetailedOutput);
	}
	
	/**
	 * PHPDOC ..
	 */
	public static function ParseSynchroImportOutput($sTrimmedOutput, $bDetailedOutput) : string
```

Reply:
I reverted the renaming.

Reply:
I know, but you could still do it, but just add this extra function to maintain backward compatibility..